pkg/util/indexeres/virtualmachine.go

```diff
@@ -0,0 +1,26 @@
+package indexeres
+
+import (
+	kubevirtv1 "kubevirt.io/api/core/v1"
+
+	"github.com/harvester/harvester/pkg/ref"
+)
+
+// The file contains the indexers which are used by controller and webhook.
+const (
+	VMByPVCIndex = "harvesterhci.io/vm-by-pvc"
+)
+
+func VMByPVC(obj *kubevirtv1.VirtualMachine) ([]string, error) {
+	var results []string
+	if obj == nil || obj.Spec.Template == nil {
+		return results, nil
+	}
+
+	for _, volume := range obj.Spec.Template.Spec.Volumes {
+		if volume.PersistentVolumeClaim != nil && volume.PersistentVolumeClaim.ClaimName != "" {
+			results = append(results, ref.Construct(obj.Namespace, volume.PersistentVolumeClaim.ClaimName))
```

Comment:
Leave a noob question here, looks like we assume that the namespace of PVCs under the VM are all the same as VM.

I'm wondering if there is a scenario that
- namesapce_a contains volume_a and vm_a
- namesapce_b contains volume_b
Can we bind volume_b (under namespace_b) to vm_a (namespace_a) ? Is this legal ?

Reply:
Yeah, from the source code, they are same namespaces.

https://github.com/kubevirt/kubevirt/blob/21b0183ea480f2084efd286b73dd5b0931c4a714/staging/src/kubevirt.io/api/core/v1/schema.go#L788-L792

Reply:
Thanks for the explanation ! :smiley: